Source/WebCore/css/CSSContentDistributionValue.cpp

```diff
@@ -36,25 +36,34 @@ CSSContentDistributionValue::CSSContentDistributionValue(CSSValueID distribution
 {
 }
 
-CSSContentDistributionValue::~CSSContentDistributionValue() = default;
+Ref<CSSContentDistributionValue> CSSContentDistributionValue::create(CSSValueID distribution, CSSValueID position, CSSValueID overflow)
+{
+    return adoptRef(*new CSSContentDistributionValue(distribution, position, overflow));
+}
 
 String CSSContentDistributionValue::customCSSText() const
 {
-    StringBuilder builder;
-    if (m_distribution != CSSValueInvalid)
-        builder.append(nameLiteral(m_distribution));
-    if (m_position != CSSValueInvalid) {
-        if (m_position == CSSValueFirstBaseline)
-            builder.append(builder.isEmpty() ? "" : " ", "first baseline");
-        else if (m_position == CSSValueLastBaseline)
-            builder.append(builder.isEmpty() ? "" : " ", "last baseline");
-        else {
-            if (m_overflow != CSSValueInvalid)
-                builder.append(builder.isEmpty() ? "" : " ", nameLiteral(m_overflow));
-            builder.append(builder.isEmpty() ? "" : " ", nameLiteral(m_position));
-        }
+    auto position = m_position;
+    auto overflow = m_overflow;
+    switch (position) {
+    case CSSValueFirstBaseline:
+        position = CSSValueFirst;
+        overflow = CSSValueBaseline;
+        break;
+    case CSSValueLastBaseline:
+        position = CSSValueLast;
+        overflow = CSSValueBaseline;
+        break;
+    default:
+        break;
     }
```

Comment:
This is kind of confusing, since `overflow` is only `safe`/`unsafe`. I think this is shorter and easier to understand:
```suggestion
    if (m_position == CSSValueFirstBaseline)
        return "first baseline"_s;
    if (m_position == CSSValueLastBaseline)
        return "last baseline"_s;
```

Reply:
I have a different idea of how to fix and rearrange. Just the `return` statement doesn’t work because it omits `distribution`.